superset/common/query_context_processor.py

```diff
@@ -463,7 +466,12 @@ def get_data(self, df: pd.DataFrame) -> str | list[dict[str, Any]]:
                     df, index=include_index, **config["CSV_EXPORT"]
                 )
             elif self._query_context.result_format == ChartDataResultFormat.XLSX:
-                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
+                ndf = df.copy()
+                columns = list(df)
+                for i, col in enumerate(columns):
+                    if coltypes[i] == GenericDataType.NUMERIC:
+                        ndf[col] = pd.to_numeric(df[col], errors="ignore")
```

Comment:
I would strongly urge adding a test case for this, as without one there's no guarantee this behavior will not break in the future.